### PR TITLE
Add TSX support

### PR DESCRIPTION
**Pull request by nickmessing**
## Description
Add TSX support to `feat/md2-btn` WIP branch.

## Motivation and Context
@johnleider asked :stuck_out_tongue:

## How Has This Been Tested?

visually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/build/webpack.prod.config.js [hunk 1]
```diff
@@ -1,5 +1,4 @@
 const merge = require('webpack-merge')
-const HappyPack = require('happypack')
```
Comment (KaelWD) [inline packages/vuetify/build/webpack.prod.config.js]: Why'd you remove happypack? It should so be removed from dependencies if there's a good reason not to use it any more.

File: packages/vuetify/build/webpack.prod.config.js [hunk 2]
```diff
@@ -30,27 +29,21 @@ module.exports = merge(baseWebpackConfig, {
   module: {
     rules: [
       {
-        test: /\.[jt]s$/,
-        use: 'happypack/loader?id=scripts',
-        exclude: /node_modules/
+        test: /\.tsx?$/,
+        exclude: /(node_modules|bower_components)/,
```
Comment (KaelWD) [inline packages/vuetify/build/webpack.prod.config.js]: We're never going to be using bower, this is pointless

File: packages/vuetify/package.json [hunk 3]
```diff
@@ -80,7 +84,7 @@
     "vue-router": "^3.0.1",
     "vue-template-es2015-compiler": "^1.6.0",
     "vuetify-loader": "^0.2.0",
-    "webpack": "^4.23.1",
+    "webpack": "^4.28.2",
```
Comment (KaelWD) [inline packages/vuetify/package.json]: I prefer to use the same version everywhere if possible, this change (and babel) should be reflected in vuetifyjs.com  too

File: packages/vuetify/build/webpack.base.config.js [hunk 4]
```diff
@@ -24,13 +21,17 @@ const cssLoaders = [
 const plugins = [
   new FriendlyErrorsWebpackPlugin({
     clearConsole: true
+  }),
+  new webpack.DefinePlugin({
+    __VUETIFY_VERSION__: JSON.stringify(vuetifyPackage.version),
+    __REQUIRED_VUE__: JSON.stringify(vuetifyPackage.peerDependencies.vue)
```
Comment (KaelWD) [inline packages/vuetify/build/webpack.base.config.js]: This has to be done with babel, the es5 and lib builds aren't run through webpack

File: packages/vuetify/package.json [hunk 5]
```diff
@@ -34,16 +34,20 @@
     "lint:fix": "concurrently -n \"tsc,tslint,eslint,types\" --kill-others-on-fail \"tsc --noEmit --pretty\" \"tslint --fix -p . -c .tslint.js src/**/*.ts -t stylish\" \"eslint --fix --ext .js,.ts src\" \"yarn test:types\""
   },
   "devDependencies": {
+    "@babel/core": "^7.2.2",
```
[the remark this answers is not in the document]
Comment (nickmessing) [reply packages/vuetify/package.json]: @KaelWD, I told @johnleider, there's a lot of cleanup to do, why is there `babel-cli` in the first place?

File: packages/vuetify/build/webpack.base.config.js (hunk 4 above)
Comment (nickmessing) [reply packages/vuetify/build/webpack.base.config.js]: @KaelWD, that babel plugin doesn't support babel 7 :(
Okay, I'll make another one

**Comment by johnleider** [comment]
Closing due to inactivity.

**Pull request closed without merging**